cmake/Modules/Findelsi.cmake

```diff
@@ -0,0 +1,34 @@
+if(NOT TARGET elsi::elsi)
+  find_package(PkgConfig QUIET)
+  pkg_check_modules(ELSI QUIET elsi)
+  if(ELSI_FOUND)
+    message(STATUS "Found 'elsi' via pkg-config")
+
+    add_library(elsi::elsi INTERFACE IMPORTED)
+    target_link_libraries(
+      elsi::elsi
+      INTERFACE
+      "${ELSI_LINK_LIBRARIES}"
+    )
+    target_include_directories(
+      elsi::elsi
+      INTERFACE
+      "${ELSI_INCLUDE_DIRS}"
+    )
+
+    if("${ELSI_LINK_LIBRARIES}" MATCHES "pexsi")
+      add_library(elsi::pexsi INTERFACE IMPORTED)
+      target_link_libraries(
+        elsi::elsi
+        INTERFACE
+        "stdc++"
+      )
```

Comment:
I think, we should not link `stdc++` directly here, as it may be called differently (or require multiple libraries) on some systems. It should be delegated to the caller to call `enable_language(cxx)` if the target `elsi::pexsi` exists. (We have to do that with the default finder as well...)

Reply:
Just an additional note on this. Calling `enable_language(cxx)` in the host `CMakeLists.txt` may be not enough, as the caller won't know, that this target must be linked with the C++ standard library. It seems that we have additionally to set the `IMPORTED_LINK_INTERFACE_LANGUAGES` property of `elsi::pexsi` to `CXX` in the finder.

Reply:
OK, this does not work as I thought it should, since the mentioned property can not be set for `INTERFACE` targets. :disappointed:  I'll play around with it more, but for the moment, probably we should go with `stdc++` in there, although that is not fully transferable...

Reply:
Observed the same thing. It would be preferable to have ELSI correctly export the dependency on the C++ libraries required for PEXSI. In DFTB+ we shouldn't have to care whether or not a C++ dependency ends up in our link line.